### ISSUE TITLE
Chapter 1 - Taskflow doesn't build

### ISSUE THREAD

**Issue by sjfricke** [PacktPublishing/3D-Graphics-Rendering-Cookbook-Second-Edition#3]
using Ubuntu 22.04 and clang-15 I get hit with

```
In file included from Chapter01/03_Taskflow/src/main.cpp:6:
deps/src/taskflow/taskflow/algorithm/for_each.hpp:93:16: error: no matching function for call to 'distance'
    size_t N = distance(beg, end, inc);
               ^~~~~~~~
deps/src/taskflow/taskflow/algorithm/for_each.hpp:167:13: note: in instantiation of function template specialization 'tf::detail::make_for_each_index_task<unsigned long long, unsigned long, unsigned long long, (lambda at Chapter01/03_Taskflow/src/main.cpp:14:65), tf::GuidedPartitioner>' requested here
    detail::make_for_each_index_task(beg, end, inc, c, std::forward<P>(part))
            ^
Chapter01/03_Taskflow/src/main.cpp:14:24: note: in instantiation of function template specialization 'tf::FlowBuilder::for_each_index<unsigned long long, unsigned long, unsigned long long, (lambda at Chapter01/03_Taskflow/src/main.cpp:14:65), tf::GuidedPartitioner>' requested here
  auto task = taskflow.for_each_index(0ull, items.size(), 1ull, [&](int i) { printf("%i", items[i]); }).name("for_each_index");
                       ^
deps/src/taskflow/taskflow/core/../utility/iterator.hpp:18:1: note: candidate template ignored: deduced conflicting types for parameter 'T' ('decay_t<unwrap_ref_decay_t<unsigned long long>>' vs. 'decay_t<unwrap_ref_decay_t<unsigned long>>')
distance(T beg, T end, T step) {
^
1 error generated.
[15/57] Building CXX object shared/CMakeFiles/SharedUtils.dir/UtilsGLTF.cpp.o
shared/UtilsGLTF.cpp:10:11: warning: enumeration values 'aiTextureMapMode_Decal' and '_aiTextureMapMode_Force32Bit' not handled in switch [-Wswitch]
  switch (mapmode[0]) {
          ^~~~~~~~~~
```

for now I just commented out that test to get the rest to build

**Comment by corporateshark** [comment]
@sjfricke Will it help if you try updating Taskflow to the latest commit [7f59941aa0d188d33704665ba6432882f6c96084](https://github.com/taskflow/taskflow/commit/7f59941aa0d188d33704665ba6432882f6c96084)? I don't have a Linux machine at hand right now to test it.

**Comment by corporateshark** [comment]
Fixed by 9d6dbb70ce044be5e57e5f65a4c3076ec7d6d1b7

**Comment by sjfricke** [comment]
worked for me, thanks
